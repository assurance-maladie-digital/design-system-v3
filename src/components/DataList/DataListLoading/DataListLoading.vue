<script lang="ts" setup>
	import { computed } from 'vue'
	import HeaderLoading from '@/components/HeaderLoading/HeaderLoading.vue'

	// Define props
	const props = defineProps({
		itemsNumber: {
			type: Number,
			default: 1,
		},
		heading: {
			type: Boolean,
			default: false,
		},
		row: {
			type: Boolean,
			default: false,
		},
	})

	const itemWidth = computed(() => (props.row ? '150px' : '90px'))
</script>

<template>
	<div
		aria-hidden="true"
		class="vd-data-list-loading"
	>
		<HeaderLoading
			v-if="heading"
			width="100px"
			height="1.5rem"
			class="mb-4"
		/>

		<ul>
			<li
				v-for="index in itemsNumber"
				:key="index + '-loading-item'"
				:class="{ 'mb-4': index !== itemsNumber }"
				class="vd-data-list-loading-item"
			>
				<HeaderLoading
					v-if="!row"
					width="60px"
					height="1rem"
					class="mb-1"
				/>
				<HeaderLoading
					:width="itemWidth"
					height="1.5rem"
				/>
			</li>
		</ul>
	</div>
</template>
