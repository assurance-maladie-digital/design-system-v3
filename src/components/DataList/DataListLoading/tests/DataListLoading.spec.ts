import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import { vuetify } from '@tests/unit/setup'

import DataListLoading from '../DataListLoading.vue'

describe('DataListLoading', () => {
	it('renders correctly', () => {
		const wrapper = mount(DataListLoading, {
			global: {
				plugins: [vuetify],
			},
			props: {
				heading: true,
				itemsNumber: 3,
				row: true,
			},
		})

		expect(wrapper.html()).toMatchSnapshot()
	})
})
