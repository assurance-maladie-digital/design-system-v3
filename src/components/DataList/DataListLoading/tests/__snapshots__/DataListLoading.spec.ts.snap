// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DataListLoading > renders correctly 1`] = `
"<div aria-hidden="true" class="vd-data-list-loading">
  <div data-v-947c5508="" class="v-skeleton-loader v-theme--light mb-4 vd-header-loading mb-4" style="height: 1.5rem; width: 100px;" arialive="polite" arialabel="Loading..." role="alert" aria-hidden="true">
    <div class="v-skeleton-loader__bone v-skeleton-loader__heading"></div>
  </div>
  <ul>
    <li class="mb-4 vd-data-list-loading-item">
      <!--v-if-->
      <div data-v-947c5508="" class="v-skeleton-loader v-theme--light vd-header-loading" style="height: 1.5rem; width: 150px;" arialive="polite" arialabel="Loading..." role="alert" aria-hidden="true">
        <div class="v-skeleton-loader__bone v-skeleton-loader__heading"></div>
      </div>
    </li>
    <li class="mb-4 vd-data-list-loading-item">
      <!--v-if-->
      <div data-v-947c5508="" class="v-skeleton-loader v-theme--light vd-header-loading" style="height: 1.5rem; width: 150px;" arialive="polite" arialabel="Loading..." role="alert" aria-hidden="true">
        <div class="v-skeleton-loader__bone v-skeleton-loader__heading"></div>
      </div>
    </li>
    <li class="vd-data-list-loading-item">
      <!--v-if-->
      <div data-v-947c5508="" class="v-skeleton-loader v-theme--light vd-header-loading" style="height: 1.5rem; width: 150px;" arialive="polite" arialabel="Loading..." role="alert" aria-hidden="true">
        <div class="v-skeleton-loader__bone v-skeleton-loader__heading"></div>
      </div>
    </li>
  </ul>
</div>"
`;
